src/pymortests/strategies.py

```diff
@@ -0,0 +1,448 @@
+def valid_inds(v, length=None):
+    if length is None:
+        yield []
+        yield slice(None)
+        yield slice(0, len(v))
+        yield slice(0, 0)
+        yield slice(-3)
+        yield slice(0, len(v), 3)
+        yield slice(0, len(v)//2, 2)
+        yield list(range(-len(v), len(v)))
+        yield list(range(int(len(v)/2)))
+        yield list(range(len(v))) * 2
+        length = 32
+    if len(v) > 0:
+        for ind in [-len(v), 0, len(v) - 1]:
+            yield ind
+        if len(v) == length:
+            yield slice(None)
+        np.random.seed(len(v) * length)
+        yield list(np.random.randint(-len(v), len(v), size=length))
+    else:
+        if len(v) == 0:
+            yield slice(0, 0)
+        yield []
+
+
+# TODO this needs to be a strategy
+def valid_inds_of_same_length(v1, v2):
+    if len(v1) == len(v2):
+        yield slice(None), slice(None)
+        yield list(range(len(v1))), list(range(len(v1)))
+        yield (slice(0, len(v1)),) * 2
+        yield (slice(0, 0),) * 2
+        yield (slice(-3),) * 2
+        yield (slice(0, len(v1), 3),) * 2
+        yield (slice(0, len(v1)//2, 2),) * 2
+    yield [], []
+    if len(v1) > 0 and len(v2) > 0:
+        yield 0, 0
+        yield len(v1) - 1, len(v2) - 1
+        yield -len(v1), -len(v2)
+        yield [0], 0
+        yield (list(range(min(len(v1), len(v2))//2)),) * 2
+        np.random.seed(len(v1) * len(v2))
+        for count in np.linspace(0, min(len(v1), len(v2)), 3).astype(int):
+            yield (list(np.random.randint(-len(v1), len(v1), size=count)),
+                   list(np.random.randint(-len(v2), len(v2), size=count)))
+        yield slice(None), np.random.randint(-len(v2), len(v2), size=len(v1))
+        yield np.random.randint(-len(v1), len(v1), size=len(v2)), slice(None)
+
+
+@hyst.composite
+def st_valid_inds_of_same_length(draw, v1, v2):
+    len1, len2 = len(v1), len(v2)
+    ret = hyst.just([([],), ([],)])
+    # TODO we should include integer arrays here
+    val1 = hynp.basic_indices(shape=(len1,), allow_ellipsis=False) #| hynp.integer_array_indices(shape=(len1,))
+    if len1 == len2:
+        ret = ret | hyst.tuples(hyst.shared(val1, key="st_valid_inds_of_same_length"), hyst.shared(val1, key="st_valid_inds_of_same_length"))
+    if len1 > 0 and len2 > 0:
+        val2 = hynp.basic_indices(shape=(len2,), allow_ellipsis=False) #| hynp.integer_array_indices(shape=(len2,))
+        ret = ret | hyst.tuples(val1, val2)#.filter()
+    # values are always tuples
+    return [d[0] for d in draw(ret)]
+
+
+@hyst.composite
+def vector_arrays_with_valid_inds_of_same_length(draw, count=2):
+    val = draw(hyst.integers(min_value=1, max_value=MAX_LENGTH))
+    length = hyst.tuples(*[hyst.just(val) for _ in range(count)])
+    vectors = draw(vector_arrays(count=count, length=length, compatible=True))
+    ind = draw(st_valid_inds_of_same_length(*vectors))
+    return vectors, ind
+
+
+# TODO this needs to be a strategy
```

Comment:
See above.

Reply:
I'd rather keep my TODOs. Why would you remove those (and below ones)?

Reply:
Well, we have never used those 'TODOs as code comments' before. I use no functionality in my editor which would make me aware of their existence. So in general I would prefer to not have them in pyMOR, also because they somewhat act as an additional issue tracker, and I am convinced that it is a bad thing to have two separate ways for tracking issues.

Having said that, I understand that with such a large refactoring, it would be tedious to have all TODOs as separate issues. So I would say, open a github issue stating that the testing related TODOs need to be fixed and leave them in.

Reply:
I can live with that :+1: